build/dial.go

```diff
@@ -0,0 +1,46 @@
+package build
+
+import (
+	"context"
+	"math/rand"
+	"net"
+
+	"github.com/docker/buildx/builder"
+	"github.com/docker/buildx/driver"
+	"github.com/docker/buildx/util/progress"
+	"github.com/pkg/errors"
+)
+
+func Dial(ctx context.Context, nodes []builder.Node, pw progress.Writer) (net.Conn, error) {
+	nodes, err := filterAvailableNodes(nodes)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(nodes) == 0 {
+		return nil, errors.New("no nodes available")
+	}
+
+	rand.Shuffle(len(nodes), func(i, j int) {
+		nodes[i], nodes[j] = nodes[j], nodes[i]
+	})
+
+	// TODO: filter nodes by platform
```

Comment:
This should be a flag for the command probably.

Reply:
Updated to support this.
Also makes the node selection use `resolveDriverBase`